modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/codegen/CodeGenerator.java

```diff
@@ -2762,8 +2757,9 @@ public void visit(BLangXMLQuotedString xmlQuotedString) {
 
     @Override
     public void visit(BLangStringTemplateLiteral stringTemplateLiteral) {
-        stringTemplateLiteral.concatExpr.regIndex = createLHSRegIndex(stringTemplateLiteral.regIndex);
```

Comment:
Can you explain why you did this change?

Reply:
This is to fix https://github.com/ballerinalang/ballerina/issues/4485.
In the prev impl, at this point,  `stringTemplateLiteral.regIndex` is not set, hence inside `createLHSRegIndex()` a null pointer is thrown.  So modeled this in the same way as \[1\]

\[1\] https://github.com/ballerinalang/ballerina/pull/4489/files/e28052fdf6c84eea1876984337ab0ad4e9fe722a#diff-5e3b169d288a55d67a94efc766adbc17L2757

Reply:
Thanks for the explanation.  I hope you've added a test case.

Reply:
Yes, its included in https://github.com/ballerinalang/ballerina/pull/4489/files/e28052fdf6c84eea1876984337ab0ad4e9fe722a#diff-fe62d65b5ce555bebf6272d23f30aa08R159